agimus_controller/agimus_controller/warm_start_base.py

```diff
@@ -12,9 +14,29 @@ def __init__(self) -> None:
     @abstractmethod
     def generate(
         self,
+        initial_state: TrajectoryPoint,
         reference_trajectory: list[TrajectoryPoint],
-    ) -> tuple(np.ndarray, np.ndarray):
-        """Returns x_init, u_init."""
+    ) -> tuple[
+        npt.NDArray[np.float64],
+        list[npt.NDArray[np.float64]],
+        list[npt.NDArray[np.float64]],
+    ]:
+        """
+        Generate initial values for a warm-start of the optimization problem.
+
+        Args:
+            initial_state (TrajectoryPoint): The initial state of the robot,
+                containing `robot_configuration` and `robot_velocity`.
+            reference_trajectory (list[TrajectoryPoint]): A list of `TrajectoryPoint` objects
+                representing the reference trajectory.
+
+        Returns:
+            tuple:
+                - x0 (npt.NDArray[np.float64]): The initial state vector.
+                - init_xs (list[npt.NDArray[np.float64]]): List of state vectors
+                for each point in the reference trajectory.
+                - init_us (list[npt.NDArray[np.float64]]): List of control inputs.
```

Comment:
```suggestion
        Args:
            initial_state: The initial state of the robot,
                containing `robot_configuration` and `robot_velocity`.
            reference_trajectory: A list of `TrajectoryPoint` objects
                representing the reference trajectory.

        Returns:
            tuple:
                - x0: The initial state vector.
                - init_xs: List of state vectors
                for each point in the reference trajectory.
                - init_us: List of control inputs.
```
Seems like google style does not require type to be specified in the doctoring.